### PR TITLE
compiler(amd64): return false on SSE4 unsupported CPUs

### LINKED ISSUE #580: Compiler(amd64): return false on SSE4 unsupported CPUs from platform.CompilerSupported
**Is your feature request related to a problem? Please describe.**

We've already used SSE4.x features in floating-point conversions and SIMD implementations. However, some _fairly old_ amd64 CPUs do not have support for these features.

Practically speaking, SSE4.1 features was introduced [15-years ago](https://en.wikipedia.org/wiki/SSE4), and actually, the SIMD proposal has been merged with the baseline being SSE4 on Intel CPUs(https://github.com/WebAssembly/simd/pull/201#issuecomment-600240445). It seems really hard (maybe impossible) to implement without the feature (https://github.com/bytecodealliance/wasmtime/issues/3809, https://github.com/bytecodealliance/wasm-micro-runtime/issues/1022,https://github.com/wasmerio/wasmer/pull/2775, https://github.com/wasmerio/wasmer/pull/2775).

It's nice if we could have support for these CPUs, but given that almost all amd64 CPUs have SSE4, that's not our priority and revisit what we can do.

**Describe the solution you'd like**

Get the available features by [`CPUID`](https://en.wikipedia.org/wiki/CPUID) instruction, and raise a compilation error if the features are necessary while not available in the hardware.

**Describe alternatives you've considered**

N/A

**Additional context**

Go has introduced `GOAMD64` flag as of [1.18](https://tip.golang.org/doc/go1.18), and we could leverage the flag and check if the flag has >=v2.

### PR DESCRIPTION
**Pull request by evacchi**
I am not sure whether we should also add some test case to exercise this code path. Maybe enabling emulated tests as suggested in #581? Even my old Window laptop is newer and passes :D

Signed-off-by: Edoardo Vacchi <evacchi@users.noreply.github.com>

Fixes #580

### CODE REVIEW

File: internal/platform/platform_arm64.go [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package platform
+
+// init verifies that the current CPU supports the required ARM64 features
+func init() {
+	// no further checks currently needed
```
Comment (mathetake) [inline internal/platform/platform_arm64.go:5]:
```suggestion
	// No further checks currently needed.
```

File: internal/platform/platform.go [hunk 2]
```diff
@@ -10,6 +10,9 @@ import (
 	"runtime"
 )
 
+// archRequirementsVerified is set by platform-specific init to true if the platform is supported
+var archRequirementsVerified = false
```
Comment (mathetake) [inline internal/platform/platform.go:14]:
```suggestion
var archRequirementsVerified bool
```

**Review by mathetake — APPROVED** [review]

**Comment by mathetake** [comment]
well, at least I would like you to verify it via QEMU, but no need to add it to the CI I think :D

**Comment by evacchi** [comment]
TBF I tried to setup QEMU and I gave up because it was so slow, but I just realized I could just download a preinstalled image instead of starting from scratch 🤦‍♂️ (probably docker would do too...)

**Comment by mathetake** [comment]
If setting toolchains locally is something too slow, you could build a binary `go test -c -o outputbin`  on your main machine, and only send the built test binary to the emulated environment. That's why you could avoid the need to install Go, Git, etc

**Comment by mathetake** [comment]
anyway merging as I _think_ this works!

**Comment by mathetake** [comment]
https://github.com/tetratelabs/wazero/pull/1122

**Comment by evacchi** [comment]
ok I confirm that it works on a qemu machine started with `-cpu qemu64,-sse4_1,-sse4_2`; i.e. `CompilerSupported()` returns false.

**Comment by mathetake** [comment]
Coolio

**Pull request merged**